### PR TITLE
move section on make variables

### PR DESCRIPTION
**Pull request by fricklerhandwerk**
# Motivation

it should be after the general build instructions, as it goes into more detail.

# Context
  - use one line per sentence
  - use imperative for instructions
  - add link to Make documentation

# Priorities and Process

Add :+1: to [pull requests you find important](https://github.com/NixOS/nix/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc).

The Nix maintainer team uses a [GitHub project board](https://github.com/orgs/NixOS/projects/19) to [schedule and track reviews](https://github.com/NixOS/nix/tree/master/maintainers#project-board-protocol).

### CODE REVIEW

File: doc/manual/src/contributing/hacking.md [hunk 1]
```diff
@@ -132,6 +111,26 @@ $ nix-build
 
 You can also build Nix for one of the [supported platforms](#platforms).
 
+## Makefile variables
+
+You may need `profiledir=$out/etc/profile.d` and `sysconfdir=$out/etc` to run `make install`.
+
+Run `make` with [`--environment-overrides`](https://www.gnu.org/software/make/manual/make.html#index-_002de) to allow environment variables to override `Makefile` variables:
+
+- `ENABLE_BUILD=yes` to enable building the C++ code.
```
Comment (roberth) [inline doc/manual/src/contributing/hacking.md:120]:
```suggestion
- `ENABLE_BUILD=yes`: build the C++ code.
```

same below

File: doc/manual/src/contributing/hacking.md [hunk 2]
```diff
@@ -132,6 +111,26 @@ $ nix-build
 
 You can also build Nix for one of the [supported platforms](#platforms).
 
+## Makefile variables
+
+You may need `profiledir=$out/etc/profile.d` and `sysconfdir=$out/etc` to run `make install`.
+
+Run `make` with [`--environment-overrides`](https://www.gnu.org/software/make/manual/make.html#index-_002de) to allow environment variables to override `Makefile` variables:
```
Comment (roberth) [inline doc/manual/src/contributing/hacking.md:118]: They don't recommend this. I think we're supposed to pass them on the command line. If your command line has a bunch of `make foo && make bar` though, I think this is indeed worth the "risk", but considering that this is just for convenience, we should suggest `-e`. That long name is way too long to use more than once in a combined shell line.

**Review by roberth — CHANGES_REQUESTED** [review]
Long name is counterproductive here.

File: doc/manual/src/contributing/hacking.md (hunk 1 above)
Comment (fricklerhandwerk) [reply doc/manual/src/contributing/hacking.md:120]: I briefly considered this, but then deemed the change too noisy, and it's not that much of an improvement in terms of clarity or readability. I can do it if you think it's a blocker.

File: doc/manual/src/contributing/hacking.md (hunk 2 above)
Comment (Ericson2314) [reply doc/manual/src/contributing/hacking.md:118]: Many things have `?=` definitions which already get overridden, too.

Comment (Ericson2314) [reply doc/manual/src/contributing/hacking.md:118]: (we replied concurrently) I think @roberth meant don't do `-e` *or* `--environment-overrides`.

File: doc/manual/src/contributing/hacking.md (hunk 1 above)
Comment (roberth) [reply doc/manual/src/contributing/hacking.md:120]: Not a blocker.

**Review by roberth — APPROVED** [review]

**Pull request merged**